container_files/assets/www/doc/builtin/lang/Javascript.md

```diff
@@ -243,4 +258,33 @@ This object has the following methods defined:
 - `dataset.type()` returns the type of the dataset
 - `dataset.config()` returns the configuration of the dataset
 - `dataset.getTimestampRange()` returns the range of timestamps present in
-  the dataset
\ No newline at end of file
+  the dataset
+
+### Procedure objects
+
+A procedure object can be created using `mldb.createProcedure(config)`.
+The `config` is the same JSON object that would be `POST`ed to `/v1/procedures`
+to create the procedure.  If no `id` is passed, then the procedure is
+anonymous and will have an `id` auto-generated.
+
+The procedure object has the following methods defined:
+
+- `procedure.run(args)` runs the procedure, with the given arguments,
+  waits for it to finish, and returns the output of the procedure.  It's
+  equivalent to `POST`ing to the `/runs` route of the procedure.
+- `procedure.id()` returns the id of the procedure
+- `procedure.type()` returns the type of the procedure
+- `procedure.config()` returns the configuration of the procedure
+
+### Function objects
+
+A function object is created similarly to procedure and dataset objects.
+
+It has the following methods defined:
+
+- `function.run(args)` runs the function, with the given arguments,
```

Comment:
It was wrong; the current word is `call`.  Apply comes from when we had the concept of "block"s, we should probably rename it as call everywhere due to us having functions now.